yeonjae/programmers-2/오픈채팅방.js

```diff
@@ -0,0 +1,29 @@
+function solution(record) {
+  const result = [];
+  const map = new Map();
+
+  for (let i = 0; i < record.length; i++) {
+    const [status, uid, nickName] = record[i].split(" ");
+
+    if (status == "Enter") {
+      result.push([uid, "님이 들어왔습니다."]);
+    }
+
+    if (status == "Leave") {
+      result.push([uid, "님이 나갔습니다."]);
+
+      continue;
+    }
+
+    map.set(uid, nickName);
+  }
+
+  return result.map((el) => map.get(el[0]) + el[1]);
+}
```

Comment:
```suggestion
function solution(record) {
  const result = [];
  const map = new Map();
  const chatType = {
    Enter: "님이 들어왔습니다.",
    Leave: "님이 나갔습니다.",
  };

  for (let i = 0; i < record.length; i++) {
    const [status, uid, nickName] = record[i].split(" ");

    if (status !== "Change") {
      result.push([uid, chatType[status]]);
    }

    if (nickName) map.set(uid, nickName);
  }

  return result.map((el) => map.get(el[0]) + el[1]);
}
```
- 풀이잘봤습니다
- 동등 연산자 실수하셨습니다 일치 연산자로 바꿔주세요 ===
- 반복적인 코드를 객체를 활용한 좀 더 깔끔하게 바꿔봤습니다!

Reply:
객체 활용한 풀이 참고하겠습니다!